skmediate/conditional_independence.py

```diff
@@ -81,30 +205,81 @@ def fit(self, X, Z, Y):
         # Step 1: Residualize with regression
 
         # TODO: Check regression type for supporting single or multi-output regression
-        regfit_xy = self.regression_estimator_xy.fit(X, Y)
-        regfit_xz = self.regression_estimator_xz.fit(X, Z)
+        if not self.residualized:
+            regfit_xz = self.regression_estimator_xz.fit(X, Z)
+            regfit_xy = self.regression_estimator_xy.fit(X, Y)
+
+            # Compute residualized Zs and Ys.
+            self.residualized_Z_ = Z - regfit_xz.predict(X)
+            self.residualized_Y_ = Y - regfit_xy.predict(X)
+        else:
+            self.residualized_Z_ = np.copy(Z)
+            self.residualized_Y_ = np.copy(Y)
 
-        # Compute residualized Zs and Ys.
-        self.residualized_Z_ = Z - regfit_xz.predict(X)
-        self.residualized_Y_ = Y - regfit_xy.predict(X)
+            if X is not None:
+                warnings.warn(
+                    "You supplied `X` to the fit method but specified "
+                    "`residualized=True` on init. This method will not use the "
+                    "`X` argument that you provided."
+                )
 
         # Step 2: Covariance estimation
-        W = np.concatenate((self.residualized_Z_, self.residualized_Y_), axis=1)
+        # Step 2a: Estimate covariance of Y,Z
+        W = np.concatenate((self.residualized_Y_, self.residualized_Z_), axis=1)
         self.covfit_zy_ = self.covariance_estimator.fit(W)
+        cols_Y = self.residualized_Y_.shape[1]
+        self.cov_zy_ = self.covfit_zy_.covariance_[:cols_Y, cols_Y:]
+
+        # Step 2b: Estimate precision of Z
+        self.covfit_zz_ = self.precision_estimator.fit(self.residualized_Z_)
+        self.prec_zz_ = self.covfit_zz_.precision_
+
+        # Step 2c: Estimate precision of Y
+        self.covfit_yy_ = self.precision_estimator.fit(self.residualized_Y_)
+        self.prec_yy_ = self.covfit_yy_.precision_
+
+        # Step 2d: Calculate residual cross-covariance
+        self.residual_crosscovariance_ = np.diag(
+            ((self.cov_zy_ @ self.prec_zz_) @ self.cov_zy_.T) @ self.prec_yy_
+        ).flatten()
+
+        n, k = Z.shape
+        self.residual_crosscovariance_wherry_corrected_ = 1 - (
+            1 - self.residual_crosscovariance_
+        ) * ((n - 1) / (n - k - 1))
+
+        if self.estimate_p_value:
+            rcc_shuffle = []
+
+            shuffle_cov_est = clone(self.covariance_estimator)
+            shuffle_prec_est = clone(self.precision_estimator)
+
+            if self.show_progress:
+                shuffle_range = trange(self.n_shuffle)
+            else:
+                shuffle_range = range(self.n_shuffle)
+
+            for n in shuffle_range:
+                shuffle_Y = shuffle(Y)
```

Comment:
I think this should be 
`shuffle_Y = shuffle(self.residualized_Y_)`

Reply:
You're right! Thanks! Fixed in latest commit.